linkedList/implementLinkedList.js

```diff
@@ -67,13 +67,15 @@ class LinkedList {
   }
 
   printList() {
-    console.log(this);
+    let node = this.head;
+    while (node) {
+      console.log(node.data)
+      node = node.next;
+    }
   }
 
-  addNode(name, data) {
-    name = new ListNode(data);
-    console.log(name);
-    list.getLast().next = name;
+  addNode(data) {
+    list.getLast().next = new ListNode(data);
   }
 }
 // Create nodes
```

Comment:
The code you have from line 80 to 90 produces the right outcome, however;

What is beneficial in using the `addNode()` function is that it abstracts (caller isnt made aware of 'how') the logic of adding a node. Ultimately the caller wants to deal with a list, and operations on that list, not the nodes necessarily.  

The `ListNode` class could be considered internal to the `LinkedList` class. Only the `LinkedList` class should call it. When a programmer wants to use a `LinkedList`, they can instantiate it to get an object and have easy to use functions:
```
let list = new LinkedList();
list.addNode(4);
list.size();
list.getLast();
..
```